### PR TITLE
Replace `chrono` with `time`

### LINKED ISSUE #115: RUSTSEC-2020-0159: Potential segfault in `localtime_r` invocations
> Potential segfault in `localtime_r` invocations

| Details             |                                                |
| ------------------- | ---------------------------------------------- |
| Package             | `chrono`                      |
| Version             | `0.4.19`                   |
| URL                 | [https://github.com/chronotope/chrono/issues/499](https://github.com/chronotope/chrono/issues/499) |
| Date                | 2020-11-10                         |

### Impact

Unix-like operating systems may segfault due to dereferencing a dangling pointer in specific circumstances. This requires an environment variable to be set in a different thread than the affected functions. This may occur without the user&#39;s knowledge, notably in a third-party library.

### Workarounds

No workarounds are known.

### References

- [time-rs/time#293](https://github.com/time-rs/time/issues/293)

See [advisory page](https://rustsec.org/advisories/RUSTSEC-2020-0159.html) for additional details.

### LINKED ISSUE #116: RUSTSEC-2020-0071: Potential segfault in the time crate
> Potential segfault in the time crate

| Details             |                                                |
| ------------------- | ---------------------------------------------- |
| Package             | `time`                      |
| Version             | `0.1.43`                   |
| URL                 | [https://github.com/time-rs/time/issues/293](https://github.com/time-rs/time/issues/293) |
| Date                | 2020-11-18                         |
| Patched versions    | `>=0.2.23`                  |
| Unaffected versions | `=0.2.0,=0.2.1,=0.2.2,=0.2.3,=0.2.4,=0.2.5,=0.2.6`               |

### Impact

Unix-like operating systems may segfault due to dereferencing a dangling pointer in specific circumstances. This requires an environment variable to be set in a different thread than the affected functions. This may occur without the user&#39;s knowledge, notably in a third-party library.

The affected functions from time 0.2.7 through 0.2.22 are:

- `time::UtcOffset::local_offset_at`
- `time::UtcOffset::try_local_offset_at`
- `time::UtcOffset::current_local_offset`
- `time::UtcOffset::try_current_local_offset`
- `time::OffsetDateTime::now_local`
- `time::OffsetDateTime::try_now_local`

The affected functions in time 0.1 (all versions) are:

- `at`
- `at_utc`
- `now`

Non-Unix targets (including Windows and wasm) are unaffected.

### Patches

Pending a proper fix, the internal method that determines the local offset has been modified to always return `None` on the affected operating systems. This has the effect of returning an `Err` on the `try_*` methods and `UTC` on the non-`try_*` methods.

Users and library authors with time in their dependency tree should perform `cargo update`, which will pull in the updated, unaffected code.

Users of time 0.1 do not have a patch and should upgrade to an unaffected version: time 0.2.23 or greater or the 0.3 series.

### Workarounds

No workarounds are known.

### References

time-rs/time#293

See [advisory page](https://rustsec.org/advisories/RUSTSEC-2020-0071.html) for additional details.

### LINKED ISSUE #384: BlockHeader timestamp field
discussion: https://github.com/FuelLabs/fuel-core/pull/345#discussion_r882809858

tldr: Replace `chrono` with `time` or `u64` inside BlockHeader.

### PR DESCRIPTION
**Pull request by ControlCplusControlV**
Chrono has been a source of issue, this moves `fuel-core` to the much lighter and more maintained `time` crate

Fixes #115 
Fixes #116 
Fixes #384

### CODE REVIEW

**Comment by Voxelot** [comment]
Looks like we still need to replace chrono in `fuel-client`: https://github.com/FuelLabs/fuel-core/blob/controlc/time/fuel-client/src/client/schema/primitives.rs#L12

**Review by mitch-fuel — APPROVED** [review]
lgtm

**Comment by ControlCplusControlV** [comment]
> Looks like we still need to replace chrono in `fuel-client`: https://github.com/FuelLabs/fuel-core/blob/controlc/time/fuel-client/src/client/schema/primitives.rs#L12

Will remove there, missed it in my first pass

**Review by leviathanbeak — CHANGES_REQUESTED** [review]
As @Voxelot already pointed, `chrono` is still referenced in `fuel-core` as well as in `fuel-relayer` and `fuel-txpool` (but only in Cargo.toml)

**Review by mitch-fuel — APPROVED** [review]
other than the above noted references lgtm
